src/m365/planner/commands/plan/plan-add.ts

```diff
@@ -0,0 +1,130 @@
+import { Logger } from '../../../../cli';
+import {
+  CommandOption
+} from '../../../../Command';
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import Utils from '../../../../Utils';
+import GraphCommand from '../../../base/GraphCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  title: string;
+  ownerGroupId?: string;
+  ownerGroupName?: string;
+}
+
+class PlannerPlanAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.PLANNER_PLAN_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a new Microsoft Planner plan';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.ownerGroupId = typeof args.options.ownerGroupId !== 'undefined';
+    telemetryProps.ownerGroupName = typeof args.options.ownerGroupName !== 'undefined';
+    return telemetryProps;
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['id', 'title', 'createdDateTime', 'owner'];
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    if (args.options.ownerGroupId) {
+      this.createPlan(args.options.ownerGroupId, args, logger, cb);
+    }
+    else {
+      this.getGroupId(args)
+        .then((groupId: string): void => {
+          this.createPlan(groupId, args, logger, cb);
+        }, (err: any): void => this.handleRejectedPromise(err, logger, cb));
+    }
+  }
+
+  private createPlan(groupId: string, args: CommandArgs, logger: Logger, cb: () => void) {
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/planner/plans`,
+      headers: {
+        'accept': 'application/json;odata.metadata=none'
+      },
+      responseType: 'json',
+      data: {
+        "owner": groupId,
+        "title": args.options.title
+      }
+    };
+    request
+      .post(requestOptions)
+      .then((res: any): void => {
+        logger.log(res);
+        cb();
+      }, (err: any): void => this.handleRejectedODataJsonPromise(err, logger, cb));
+  }
+
+  private getGroupId(args: CommandArgs): Promise<string> {
+
+    const requestOptions: any = {
+      url: `${this.resource}/v1.0/groups?$filter=displayName eq '${encodeURIComponent(args.options.ownerGroupName as string)}'`,
+      headers: {
+        accept: 'application/json;odata.metadata=none'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<{ value: any[] }>(requestOptions)
```

Comment:
Since we're expecting the return object to contain the `id` property, let's define the response as `{id:string}[]` rather than `any[]` to avoid refactoring issues in the future.